spec/lib/lending/collector_spec.rb

```diff
@@ -84,25 +84,44 @@ def expect_to_process(item_dirname)
         expect(collector.stopped?).to eq(false)
       end
 
+      # rubocop:disable RSpec/ExampleLength
       it 'finds the next file' do
         processing_dirs = []
         final_dirs = []
 
-        expect(BerkeleyLibrary::Logging.logger).to receive(:info).with(/starting/).ordered
+        logger = BerkeleyLibrary::Logging.logger
+
+        # Collect all logs in an array
+        logs = []
+
+        # Allow all other info calls to pass through unchanged
+        allow(logger).to receive(:info).and_call_original
+
+        # Capture only the two specific log lines we assert on
+        allow(logger).to receive(:info).with(/starting|nothing left to process/) do |msg|
+          logs << msg
+        end
 
         %w[b12345678_c12345678 b86753090_c86753090].each do |item_dir|
           pdir, fdir = expect_to_process(item_dir)
           processing_dirs << pdir
           final_dirs << fdir
         end
 
-        expect(BerkeleyLibrary::Logging.logger).to receive(:info).with(/nothing left to process/).ordered
```

Comment:
do we know why there was this specific expectation here? i just want to make sure that i'm not missing something important that the spec was trying to catch.